server/pulp/server/webservices/views/roles.py

```diff
@@ -0,0 +1,229 @@
+import json
+
+from django.core.urlresolvers import reverse
+from django.utils.encoding import iri_to_uri
+from django.views.generic import View
+from django.http import HttpResponseRedirect
+from pulp.server import exceptions as pulp_exceptions
+from pulp.server.auth import authorization
+from pulp.server.managers import factory
+from pulp.server.webservices.controllers.base import json_encoder as pulp_json_encoder
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views.util import (generate_json_response,
+                                                generate_json_response_with_pulp_encoder)
+
+
+class RolesView(View):
+    """
+    Views for roles.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        List all roles.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :return: Response containing a list of roles
+        :rtype: django.http.HttpResponse
+        """
+        role_query_manager = factory.role_query_manager()
+        user_query_manager = factory.user_query_manager()
+        permissions_manager = factory.permission_manager()
+        roles = role_query_manager.find_all()
+        for role in roles:
+            role['users'] = [u['login'] for u in
+                             user_query_manager.find_users_belonging_to_role(role['id'])]
+
+            resource_permission = {}
+            # isolate schema change
+            if role['permissions']:
+                for item in role['permissions']:
+                    resource = item['resource']
+                    operations = item.get('permission', [])
+                    resource_permission[resource] = [permissions_manager.operation_value_to_name(o)
+                                                     for o in operations]
+
+            role['permissions'] = resource_permission
+        for role in roles:
```

Comment:
You already iterate through the ``roles`` on line 35, so do you actually need a second pass of iteration here? Or can you put the steps below in the loop above?